src/Accounts.js

```diff
@@ -0,0 +1,188 @@
+      this.getDate(testDate)
+    );
+    this.#transactions.push(newTransaction);
+  }
+
+  debit(amount, testDate) {
+    if (this.balance - amount < 0 - this.#overdraft) {
+      throw new Error("Insufficient funds");
+    }
+    const newTransaction = new Debit(
+      numeral(amount).format("0.00"),
+      this.getDate(testDate)
+    );
+    this.#transactions.push(newTransaction);
+  }
+
+  getDate(testDate) {
+    if (testDate) {
+      const date = new Date(testDate);
+      return date;
+    }
+    const date = new Date();
+    return date;
+  }
+
+  getStatement(type, startDate, endDate) {
+    let thisStatement;
+
+    if (startDate && endDate) {
+      const statementStartDate = new Date(startDate);
+      const statementEndDate = new Date(endDate);
+
+      if (
+        isNaN(statementStartDate.getTime()) ||
+        isNaN(statementEndDate.getTime())
+      ) {
+        throw new Error("Date format must be YYYY-MM-DD");
+      }
+
+      thisStatement = new Statement(this, statementStartDate, statementEndDate);
+    } else {
+      thisStatement = new Statement(this);
+    }
+
+    if (type === "JSON") {
+      return thisStatement.json;
+    }
+    if (type === "pdf") {
+      return thisStatement.getPDF();
+    }
+    if (type === "Console") {
+      thisStatement.console;
+      return;
+    }
+    return thisStatement;
+  }
+
+  getTransactions(startDate, endDate) {
+    const transactionsToSort = [...this.#transactions];
+
+    const transactionsSortedByDate = transactionsToSort.sort(
+      (a, b) => a.date.getTime() - b.date.getTime()
+    );
+    let ongoingBalance = 0;
+
+
+
+    transactionsSortedByDate.forEach((transaction) => {
+      if (transaction.constructor.name === "Debit") {
+        transaction.balanceAfterTransaction = numeral(
+          ongoingBalance - Number(transaction.amount)
+        ).format("0.00");
+        ongoingBalance -= Number(transaction.amount);
+      }
+      if (transaction.constructor.name === "Credit") {
+        transaction.balanceAfterTransaction = numeral(
+          ongoingBalance + Number(transaction.amount)
+        ).format("0.00");
+        ongoingBalance += Number(transaction.amount);
+      }
+    });
+
+    if (startDate && endDate) {
+      const transactionsWithinPeriod = transactionsSortedByDate.filter(
+        (transaction) =>
+          transaction.date > new Date(startDate) &&
+          transaction.date < new Date(endDate)
+      );
+      return transactionsWithinPeriod;
+    }
+
+    return transactionsSortedByDate;
+  }
+
+  get balance() {
+    const totalCredit = this.credits.reduce((a, b) => a + Number(b.amount), 0);
+    const totalDebit = this.debits.reduce((a, b) => a + Number(b.amount), 0);
+
+    return totalCredit - totalDebit;
+  }
+
+  get credits() {
+    const filteredTransactions = this.#transactions.filter(
+      (transaction) => transaction.constructor.name === "Credit"
+    );
+    return [...filteredTransactions];
```

Comment:
`filter` will return a new array containing the filtered items so you're safe to return it directly

```js
return this.#transactions.filter(
      (transaction) => transaction.constructor.name === "Credit"
    )
```